x-pack/platform/plugins/shared/streams_app/public/components/stream_detail_significant_events_view/add_significant_event_flyout/common/preview_data_spark_plot.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { niceTimeFormatter } from '@elastic/charts';
+import { EuiFlexGroup, EuiPanel, EuiText } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import type { StreamQueryKql, Streams } from '@kbn/streams-schema';
+import React, { useMemo } from 'react';
+import type { TimeRange } from '@kbn/es-query';
+import { useTimefilter } from '../../../../hooks/use_timefilter';
+import { SparkPlot } from '../../../spark_plot';
+import { useSignificantEventPreviewFetch } from '../manual_flow_form/use_significant_event_preview_fetch';
+import { useSparkplotDataFromSigEvents } from '../manual_flow_form/use_spark_plot_data_from_sig_events';
+
+export function PreviewDataSparkPlot({
+  query,
+  definition,
+  timeRange,
+  isQueryValid,
+}: {
+  definition: Streams.all.Definition;
+  query: StreamQueryKql;
+  timeRange: TimeRange;
+  isQueryValid: boolean;
+}) {
+  const {
+    timeState: { start, end },
+  } = useTimefilter();
+
+  const previewFetch = useSignificantEventPreviewFetch({
+    name: definition.name,
+    kqlQuery: query.kql.query,
+    timeRange,
+    isQueryValid,
+  });
+
+  const xFormatter = useMemo(() => {
+    return niceTimeFormatter([start, end]);
+  }, [start, end]);
+
+  const sparkPlotData = useSparkplotDataFromSigEvents({
+    previewFetch,
+    queryValues: query,
+    xFormatter,
+  });
+
+  return (
+    <EuiPanel hasBorder={true}>
+      <EuiFlexGroup direction="column" gutterSize="s">
+        <EuiText size="xs">
+          <h4>
+            {i18n.translate(
+              'xpack.streams.addSignificantEventFlyout.manualFlow.previewChartTitle',
+              { defaultMessage: 'Occurrences' }
+            )}
+          </h4>
+        </EuiText>
+        <SparkPlot
```

Comment:
The chart can be empty when the documents found to generate the query are outside of the preview window, not sure what we can do here besides using the _generate window in the chart but I assume it would be too expensive
Also in that view there is no indicator of the time range we're looking at besides hovering the graph, can we add an x axis legend without eating too much space ?